src/types/infer.ts

```diff
@@ -2298,14 +2298,59 @@ export class InferType {
         return specializeType(this.typeNameToTypeNode(astT), loc);
     }
 
+    /**
+     * Determine if the specified type `typ` is dynamic or not. Dynamic means
+     * that if we are trying to read `typ` at location `loc`, in `loc` there should be just a
+     * uint256 offset into memory/storage/calldata, where the actual data lives. Otherwise
+     * (if the type is "static"), the direct encoding of the data will start at `loc`.
+     *
+     * Usually "static" types are just the value types - i.e. anything of statically
+     * known size that fits in a uint256. As per https://docs.soliditylang.org/en/latest/abi-spec.html#formal-specification-of-the-encoding
+     * there are several exceptions to the rule when encoding types in calldata:
+     *
+     * 1. Fixed size arrays with fixed-sized element types
+     * 2. Tuples where all the tuple elements are fixed-size
+     *
+     * @todo (Dimo):
+     * 1. Check again that its not possible for tuples in internal calls to somehow get encoded on the stack
+     * 2. What happens with return tuples? Are they always in memory?
+     */
+    isTypeEncodingDynamic(typ: TypeNode): boolean {
```

Comment:
Could we rename this function to `isABITypeEncodingDynamic` just to make it clear that it is only talking about the ABI types?